ACS commons - Breaking the AEM cloud CI/CD Pipeline - AEM cloud build failure

We are encountering a deployment issue with ACS Commons version 6.11.0 on AEM as a Cloud Service. During the deployment process, we are seeing the following error in the logs:

[ERROR] com.adobe.granite.fact.command.CompileJavaCommand - Errors:
apps/acs__002d__commons/renderconditons/iscurrentusermemberof/iscurrentusermemberof__002e__jsp.java 61 log cannot be resolved
Affected

AEM Versions:
2025.3.19823.20250304T101418Z

2025.4.20476.20250414T214138Z

We had raised this with Adobe Support, but they have redirected us to report the issue here on the ACS Commons repository.

Could you please help us with a fix or workaround for this issue?

Thank you for your support.

Thank you @YegorKozlov
 
Even I see below errors in logs:

[2025/04/16 17:03:10.164] [ERROR] com.adobe.granite.fact.command.CompileJavaCommand - Errors:
apps/acs__002d__commons/components/utilities/twitter__002d__rate__002d__limit__002d__checker/body__002e__jsp.java 170 The type twitter4j.Twitter cannot be resolved. It is indirectly referenced from required .class files
apps/acs__002d__commons/components/utilities/twitter__002d__rate__002d__limit__002d__checker/body__002e__jsp.java 170 The method getTwitter() from the type TwitterClient refers to the missing type Twitter

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,6 +10,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com)
 
 ## Unreleased ([details][unreleased changes details])
 - #3537 Content Sync: preserve mix:referenceable mixin on Assets and Content Fragments
+- #3562 Fixed compilation errors in iscurrentusermemberof render condition
 
  ## 6.11.0 - 2025-03-14
 

--- a/ui.apps/src/main/content/jcr_root/apps/acs-commons/renderconditons/iscurrentusermemberof/iscurrentusermemberof.jsp
+++ b/ui.apps/src/main/content/jcr_root/apps/acs-commons/renderconditons/iscurrentusermemberof/iscurrentusermemberof.jsp
@@ -74,7 +74,7 @@
             final Authorizable authorizable = resourceResolver.adaptTo(Authorizable.class);
             final List<String> groupsList = Arrays.asList(groups);
             isAdmin = isUserAdmin(authorizable);
-            if (isUserMemberOf(authorizable, groupsList)) {
+            if (isUserMemberOf(authorizable, groupsList, log)) {
                 testcondition = true;
             }
         } catch (Exception ex) {
@@ -110,7 +110,7 @@
         return false;
     }
 
-    public static boolean isUserMemberOf(Authorizable authorizable, List<String> groups) {
+    public static boolean isUserMemberOf(Authorizable authorizable, List<String> groups, org.slf4j.Logger log) {
         try {
             Iterator<Group> groupIt = authorizable.memberOf();
             while (groupIt.hasNext()) {
@@ -120,7 +120,7 @@
                 }
             }
         } catch (Exception ignored) {
-            // log.debug("Ignored exception occurred while checking iscurrentusermemberof.jsp isUserMemberOf(authorizable, groups)", ignored.getMessage());
+            log.debug("Ignored exception occurred while checking iscurrentusermemberof.jsp isUserMemberOf(authorizable, groups)", ignored.getMessage());
         }
 
         return false;

--- a/ui.apps/src/main/content/jcr_root/apps/acs-commons/renderconditons/iscurrentusermemberof/iscurrentusermemberof.jsp
+++ b/ui.apps/src/main/content/jcr_root/apps/acs-commons/renderconditons/iscurrentusermemberof/iscurrentusermemberof.jsp
@@ -120,7 +120,7 @@
                 }
             }
         } catch (Exception ignored) {
-            log.debug("Ignored exception occurred while checking iscurrentusermemberof.jsp isUserMemberOf(authorizable, groups)", ignored.getMessage());
+            // log.debug("Ignored exception occurred while checking iscurrentusermemberof.jsp isUserMemberOf(authorizable, groups)", ignored.getMessage());
         }
 
         return false;

